pkg/config/replica_config.go

```diff
@@ -56,11 +58,12 @@ var defaultReplicaConfig = &ReplicaConfig{
 			Delimiter:  Comma,
 			NullString: NULL,
 		},
-		EncoderConcurrency:       16,
+		EncoderConcurrency:       util.AddressOf(16),
 		Terminator:               CRLF,
 		DateSeparator:            DateSeparatorNone.String(),
-		EnablePartitionSeparator: true,
-		EnableKafkaSinkV2:        false,
+		EnablePartitionSeparator: util.AddressOf(true),
```

Comment:
Should the default value here be set in `validateAndAdjust` according to sink type?  eg, set `EnablePartitionSeparator` to true only when using storage sink.